app/Repositories/EloquentRepository.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Repositories\RepositoryInterface;
+
+abstract class EloquentRepository implements RepositoryInterface
+{
+    protected $_model;
+
+    public function __construct()
+    {
+        $this->setModel();
+    }
+
+    abstract public function getModel();
+
+    public function setModel()
+    {
+        $this->_model = app()->make(
+            $this->getModel()
+        );
+    }
+
+    public function getAll()
+    {
+        return $this->_model->all();
+    }
+
+    public function getPaginate()
+    {
+        return $this->_model->latest()->paginate(config('custom.PagePaginate'));
+    }
+
+    public function getLatest()
+    {
+        return $this->_model->latest();
+    }
+
+    public function getWith(array $attributes)
+    {
+        $result = $this->_model->with($attributes)->get();
+
+        return $result;
+    }
```

Comment:
function này có ý nghĩa gì?

Reply:
dùng để lấy ra những field cần thiết thay vì lấy toàn bộ thì truyền những field cần lấy ra vào attributes để lấy